apps/interactor/src/state/listeners.ts

```diff
@@ -11,15 +11,36 @@ import { RootState } from './store'
 import textCompletion from '../libs/textCompletion'
 import PromptBuilder from '../libs/memory/PromptBuilder'
 
+let lastTime = new Date().getTime()
+
 const interactionEffect = async (
   dispatch: Dispatch,
   state: RootState,
   servicesEndpoint: string
 ) => {
   try {
+    const response = await fetch(
+      `${servicesEndpoint}/text/strategy/${state.settings.model}`,
+      {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        cache:
+          new Date().getTime() - lastTime > 600000 ? 'reload' : 'force-cache',
+      }
+    )
+    lastTime = new Date().getTime()
+
+    if (!response.ok || response.status != 200) {
+      throw new Error('Error getting prompt strategy')
+    }
+
+    const data = await response.json()
+
     const promptBuilder = new PromptBuilder({
       maxNewTokens: 200,
-      strategy: 'roleplay',
+      strategy: data.strategy,
```

Comment:
Maybe to be safe we could have something like
```suggestion
      strategy: isOfTypeStrategySlug(data.strategy) ? data.strategy : 'alpacarp',
```

Reply:
added in latest commit